### PR TITLE
UI restyle

### PR DESCRIPTION
**Pull request by MisterDA**
Serve js and css from static files.

### CODE REVIEW

**Comment by MisterDA** [comment]
Supersedes and closes #258.

**Comment by samoht** [comment]
Could you rebase this to master?

**Review by tmcgilchrist — COMMENTED** [review]
Can you rebase and try a deploy of it to ocaml-ci.

**Comment by MisterDA** [comment]
I think this is ready, please let me know what you think.
The PR builds on the UI restyle by Ewan Mellor, fixes a few content-type bugs, serves files "statically" (which means they need to be installed on the host filesystem).
From the base ui-restyle PR, it removes the hovering animations and doesn't use the Roboto webfont, but rather the preferred system font.

Do you think static files should be served with something like ocaml-crunch instead? I think it's better either way not to embed them in the source, we get syntax highlighting and linting.

**Comment by MisterDA** [comment]
This PR supersedes and closes #339.

**Comment by tmcgilchrist** [comment]
Note that this needs updating in https://github.com/ocurrent/ocaml-multicore-ci/pull/22 so it can switch to mainline ocurrent.

File: lib_web/resource.ml [hunk 1]
```diff
@@ -94,3 +94,42 @@ let logout = object
...
+          ("Content-Type", content_type);
+          ("Cache-Control", Printf.sprintf "public, max-age=%d;" max_age);
+        ]
+    in
+    Utils.Server.respond_string ~status:`OK ~headers ~body ()
+  end
+
+(* Serve a static asset from a resource embedded with ocaml-crunch,
+   with a cache of one day (by default). *)
+let crunch ?content_type ?(max_age=86400) _path = object
```
Comment (tmcgilchrist) [inline lib_web/resource.ml:117]:
```suggestion
let crunch ?content_type ?(max_age=86400) _ = object
```

**Review by tmcgilchrist — APPROVED** [review]

**Comment by MisterDA** [comment]
Applied your suggestion, thanks.

**Pull request merged**